### PR TITLE
Waiting foor your feedback

### PR DESCRIPTION
**Pull request by VadymDemchenko**

### CODE REVIEW

File: HW18/homework_18.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        self.url = None
+
+    def set_url(self, url=None):
+        self.url = url
+
+    def set_chat_id(self, chat_id=None):
+        self.chat_id = chat_id
+
+    def send_message(self, message):
+        print(f"{self.name} bot says {message} to chat {self.chat_id} using url {self.set_url()}")
```
Comment (klymenok) [inline HW18/homework_18.py]: self.set_url() will reset your url even if you set it before and won't return any value to print. Maybe you meant `self.url`?

File: HW18/homework_18.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+telegram_bot.send_message('Hello')
+
+telegram_bot.set_chat_id(1)
+
+telegram_bot.send_message('Hello')
+
+
+class MyStr(str):
+    def __str__(self):
+        return super().__str__().upper()
```
Comment (klymenok) [inline HW18/homework_18.py]: You can simplify this to `return self.upper()` since `self` is already a string

File: HW18/homework_18.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+
+my_str = MyStr("test")
+print(my_str)
+
+
+class User:
+    def __init__(self, name):
+        self.name = name.lower()
```
Comment (klymenok) [inline HW18/homework_18.py]: This part is changing an object and making its name in lowercase, but the goal is not to change the initial value but to change the behavior for comparison in `__eq__` method.

File: HW18/homework_18.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+    def __eq__(self, other_name):
+        return self.name == other_name.name
```
Comment (klymenok) [inline HW18/homework_18.py]: Compare here `name` transformed to lowercase to reach the goal of the task

**Review by klymenok — CHANGES_REQUESTED** [review]

File: HW18/homework_18.py (hunk 1 above)
Comment (VadymDemchenko) [reply HW18/homework_18.py]: yes, you're right. I meant self.url

File: HW18/homework_18.py (hunk 2 above)
Comment (VadymDemchenko) [reply HW18/homework_18.py]: Wow. Really... Thanks! :)

File: HW18/homework_18.py (hunk 4 above)
Comment (VadymDemchenko) [reply HW18/homework_18.py]: Got that. Thanks. I missed the point that we can't change original value

**Review by klymenok — APPROVED** [review]
Looks good!

**Pull request merged**